libs/sdk-ui-kit/src/StylingPicker/StylingPicker.tsx

```diff
@@ -0,0 +1,196 @@
+export interface IStylingPickerProps {
+    title: string;
+    basicItem: StylingPickerItem;
+    customItems: StylingPickerItem[];
+    emptyMessageElement: JSX.Element;
+    selectedItemRef?: ObjRef;
+    isLoading?: boolean;
+    titleTooltip?: string;
+    footerHelpLink?: string;
+    footerHelpTitle?: string;
+    footerMobileMessage?: string;
+    className?: string;
+    onApply?: (ref: ObjRef) => void;
+    onListActionClick?: () => void;
+    locale?: string;
+}
+
+const StylingPickerCore: React.FC<IStylingPickerProps & WrappedComponentProps> = (props) => {
+    const {
+        intl,
+        title,
+        basicItem,
+        customItems,
+        emptyMessageElement,
+        selectedItemRef,
+        isLoading,
+        titleTooltip,
+        footerHelpLink,
+        footerHelpTitle,
+        footerMobileMessage,
+        onApply,
+        onListActionClick,
+        className,
+    } = props;
+    const [currentItemRef, setCurrentItemRef] = useState<ObjRef>(selectedItemRef || null);
+    const isMobileDevice = useMediaQuery("mobileDevice");
+
+    useEffect(() => {
+        setCurrentItemRef(selectedItemRef);
+    }, [selectedItemRef]);
+
+    const onItemClick = useCallback((ref: ObjRef) => {
+        setCurrentItemRef(ref);
+    }, []);
+
+    const showFooterButtons = useMemo(
+        () => customItems.length > 0 && (selectedItemRef || currentItemRef),
+        [customItems, selectedItemRef, currentItemRef],
+    );
+
+    const isApplyButtonDisabled = useMemo(
+        () => areObjRefsEqual(currentItemRef, selectedItemRef),
+        [currentItemRef, selectedItemRef],
+    );
+
+    const handleCancel = useCallback(() => {
+        setCurrentItemRef(selectedItemRef);
+    }, [selectedItemRef]);
+
+    const handleApply = useCallback(() => {
+        onApply?.(currentItemRef);
+    }, [onApply, currentItemRef]);
+
+    return (
+        <div className={cx("gd-styling-picker s-styling-picker", className)}>
+            <div className="gd-styling-picker-title s-styling-picker-title">
+                <BubbleHoverTrigger showDelay={0} hideDelay={0}>
+                    <div className="gd-styling-picker-title-content">
+                        <Typography tagName="h3">{title}</Typography>
+                        {titleTooltip && (
+                            <span className="gd-icon-circle-question gd-styling-picker-title-icon" />
+                        )}
+                    </div>
+                    {titleTooltip && (
+                        <Bubble className="bubble-primary" alignPoints={TOOLTIP_ALIGN_POINTS}>
+                            {titleTooltip}
+                        </Bubble>
+                    )}
+                </BubbleHoverTrigger>
+            </div>
+            <div className="gd-styling-picker-content">
+                {isLoading ? (
+                    <DialogListLoading className="gd-styling-picker-list-loading s-styling-picker-list-loading" />
+                ) : (
+                    <>
```

Comment:
Optional: split this to multiple subcomponents to make it more readable (eg StylingPickerHeader / StylingPickerBody / StylingPickerFooter)